### PR TITLE
Allow you to stop pulling by pulling the same object again.

### PR DESCRIPTION
**Pull request by neersighted**
:cl: neersighted
tweak: You can now stop pulling by trying to pull the same object again.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
Neat :+1:

**Comment by RemieRichards** [comment]
This is one of those things we've been missing but never knew :+1:

**Comment by Giacom** [comment]
I swear this was a thing before.

**Comment by MrStonedOne** [comment]
Honestly, any control click that doesn't lead to pulling should stop pulling.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -162,6 +162,8 @@ Sorry Giacom. Please don't be mad :(
 
 	if(AM.Adjacent(src))
 		src.start_pulling(AM)
+	if(pulling == AM)
+		stop_pulling()
```
Comment (phil235) [inline code/modules/mob/living/living.dm]: Won't this make pulling impossible? pullin == AM will always be true if the start_pulling(AM) just above succeeds so we immediately stop pulling the thing we started to pull.

**Comment by AlexanderUlanH** [comment]
We had this and I liked it, but when it was taken away people were happy about it, so
:+1: :-1:

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -160,6 +160,8 @@ Sorry Giacom. Please don't be mad :(
 	set name = "Pull"
 	set category = "Object"
 
+	if(pulling == AM)
+		stop_pulling()
```
Comment (phil235) [inline code/modules/mob/living/living.dm]: now you have the opposite issue, if you click the object  you're currently pulling then you will stop pulling it and immediately start pulling it again.
I think you should change `if(AM.Adjacent(src))` to `else if(AM.Adjacent(src))`.

**Pull request merged**